### ISSUE TITLE
The Surprising Effectiveness of Test-Time Training for Abstract Reasoning, 2024.11

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#1500]
https://ekinakyurek.github.io/papers/ttt.pdf

**Comment by AkihikoWatanabe** [comment]
__translate:Language models have shown impressive performance on tasks within their training distri-
bution, but often struggle with novel problems requiring complex reasoning. We investigate
the effectiveness of test-time training (TTT)—updating model parameters temporarily dur-
ing inference using a loss derived from input data—as a mechanism for improving models’
reasoning capabilities, using the Abstraction and Reasoning Corpus (ARC) as a benchmark.
Through systematic experimentation, we identify three crucial components for successful
TTT: (1) initial finetuning on similar tasks (2) auxiliary task format and augmentations (3)
per-instance training. TTT significantly improves performance on ARC tasks, achieving up
to 6×improvement in accuracy compared to base fine-tuned models; applying TTT to a
8B-parameter language model, we achieve 53% accuracy on the ARC’s public validation set,
improving the state-of-the-art by nearly 25% for public and purely neural approaches. By
ensembling our method with recent program generation approaches, we get SoTA public
validation accuracy of 61.9%, matching the average human score. Our findings suggest
that explicit symbolic search is not the only path to improved abstract reasoning in neural
language models; additional test-time applied to continued training on few-shot examples
can also be extremely effective.
